scylla-cql/src/frame/types.rs

```diff
@@ -174,10 +180,19 @@ pub fn read_short(buf: &mut &[u8]) -> Result<i16, ParseError> {
     Ok(v)
 }
 
+pub fn read_u16(buf: &mut &[u8]) -> Result<u16, ParseError> {
+    let v = buf.read_u16::<BigEndian>()?;
+    Ok(v)
+}
+
 pub fn write_short(v: i16, buf: &mut impl BufMut) {
     buf.put_i16(v);
 }
 
+pub fn write_u16(v: u16, buf: &mut impl BufMut) {
+    buf.put_u16(v);
+}
```

Comment:
Maybe we should change `write_short` to accept `u16`.
AFAIU `short` refers to the CQL `short` type, which is a `u16`.
But we can do that in a separate PR as it will be a large change. Let's keep this one focused on the problem with Batches.

Reply:
I agree about changing `write_short` to accept `u16`. There is only a small number of uses of this function in the code (I counted only six) so it should be easy to fix it right now.

Reply:
The switch wasn't so simple, i had to change the type in some other structs that were linked in a way I don't understand yet. [example](https://github.com/scylladb/scylla-rust-driver/blob/8dfdcdbc6ea7299b76ee79e625f4a4f8cd009f15/scylla-cql/src/frame/value.rs#L66). But as long as the tests pass, it should be fine.